### PR TITLE
Add utilties to make it easy to deploy DJL model to SageMaker

### PR DESCRIPTION
**Pull request by frankfliu**
## Description ##

Brief description of what this PR is about

- If this change is a backward incompatible change, why must this change be made?
- Interesting edge cases to note here

### CODE REVIEW

File: extensions/aws-ai/src/test/java/ai/djl/aws/sagemaker/SageMakerTest.java [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+                Criteria.builder()
+                        .setTypes(NDList.class, NDList.class)
+                        .optModelUrls("https://resources.djl.ai/test-models/mlp.tar.gz")
+                        .build();
+        try (ZooModel<NDList, NDList> model = ModelZoo.loadModel(criteria)) {
+            SageMaker sageMaker =
+                    SageMaker.builder()
+                            .setModel(model)
+                            .optBucketName("djl-sm-test")
+                            .optContainerImage("125045733377.dkr.ecr.us-east-1.amazonaws.com/djl")
```
Comment (lanking520) [inline extensions/aws-ai/src/test/java/ai/djl/aws/sagemaker/SageMakerTest.java:56]: Is this image public accessible?

File: extensions/aws-ai/src/main/java/ai/djl/aws/sagemaker/SageMaker.java [hunk 2]
```diff
@@ -0,0 +1,670 @@
...
+        waiter.waitUntilEndpointInService(describeReq);
+        logger.info("SageMaker endpoint {} created: {}", endpointName, endpointArn);
+    }
+
+    /** Deletes the Amazon SageMaker endpoint. */
+    public void deleteEndpoint() {
+        logger.info("Deleting SageMaker endpoint {} ...", endpointName);
+        DeleteEndpointRequest req =
+                DeleteEndpointRequest.builder().endpointName(endpointName).build();
+        sageMaker.deleteEndpoint(req);
```
Comment (lanking520) [inline extensions/aws-ai/src/main/java/ai/djl/aws/sagemaker/SageMaker.java:151]: Any cases this action might be failed?

File: extensions/aws-ai/src/main/java/ai/djl/aws/sagemaker/SageMaker.java [hunk 3]
```diff
@@ -0,0 +1,670 @@
...
+        instanceCount = builder.instanceCount;
+        region = DefaultAwsRegionProviderChain.builder().build().getRegion();
+    }
+
+    /**
+     * Deploys the model to Amazon SageMaker hosting service.
+     *
+     * @throws IOException if failed upload model to S3
+     */
+    public void deploy() throws IOException {
```
Comment (lanking520) [inline extensions/aws-ai/src/main/java/ai/djl/aws/sagemaker/SageMaker.java:125]: Maybe we can set something like below:

```
deploy(SageMaker.END_POINT);
```
There could be SageMaker.END_POINT, SageMaker.LAMBDA, SageMaker.ECS...

File: extensions/aws-ai/src/main/java/ai/djl/aws/sagemaker/SageMaker.java [hunk 4]
```diff
@@ -0,0 +1,670 @@
...
+
+        Path dir = model.getModelPath();
+        Path tarFile = tar(dir);
+        String modelNameKey;
+        if (bucketPath.isEmpty()) {
+            modelNameKey = modelName + ".tar.gz";
+        } else {
+            modelNameKey = bucketPath + '/' + modelName + ".tar.gz";
+        }
+        String url = uploadModel(bucketName, modelNameKey, tarFile);
```
Comment (lanking520) [inline extensions/aws-ai/src/main/java/ai/djl/aws/sagemaker/SageMaker.java:258]: Maybe we can allow user to do this step? And pass the S3 bucket link if needed

Comment (lanking520) [reply extensions/aws-ai/src/main/java/ai/djl/aws/sagemaker/SageMaker.java:258]: Since we already had S3 Plugin, some of these functions can be placed there.

File: extensions/aws-ai/src/main/java/ai/djl/aws/sagemaker/SageMaker.java (hunk 3 above)
Comment (goswamig) [reply extensions/aws-ai/src/main/java/ai/djl/aws/sagemaker/SageMaker.java:125]: What is `SageMaker.LAMBDA`  or `SageMaker.ECS`?

File: extensions/aws-ai/src/main/java/ai/djl/aws/sagemaker/SageMaker.java [hunk 5]
```diff
@@ -0,0 +1,670 @@
...
+    private String bucketName;
+    private String bucketPath;
+    private String executionRole;
+    private String containerImage;
+    private String endpointConfigName;
+    private String endpointName;
+    private String instanceType;
+    private int instanceCount;
+
+    private SageMaker(Builder builder) {
```
Comment (goswamig) [inline extensions/aws-ai/src/main/java/ai/djl/aws/sagemaker/SageMaker.java:102]: Where is this `Builder` type define?

File: extensions/aws-ai/src/main/java/ai/djl/aws/sagemaker/SageMaker.java (hunk 2 above)
Comment (goswamig) [reply extensions/aws-ai/src/main/java/ai/djl/aws/sagemaker/SageMaker.java:151]: probably yes if the endpoint does not exist.

Comment (goswamig) [reply extensions/aws-ai/src/main/java/ai/djl/aws/sagemaker/SageMaker.java:151]: same comment is applicable to next few functions.

**Review by lanking520 — APPROVED** [review]
